File: oop/heritage/CPP/exercice3/src/Document.cpp
#include "Document.hpp"

Document::Document() : numEnregistrement(0), titre("") {}

Document::Document(int num, const std::string& titre) : numEnregistrement(num), titre(titre) {}

int Document::getNumEnregistrement() const {
    return numEnregistrement;
}

void Document::setNumEnregistrement(int newNumEnregistrement) {
    numEnregistrement = newNumEnregistrement;
}

const std::string& Document::getTitre() const {
    return titre;
}

void Document::setTitre(const std::string& newTitre) {
    titre = newTitre;
}

std::string Document::toString() const {
    return "Document [numEnregistrement=" + std::to_string(numEnregistrement) + ", titre=" + titre + "]";
}

File: oop/heritage/Java/execice 3/src/Document.java
public class Document {
	private int numEnregistrement;
    private String titre;
    public Document() {
    	numEnregistrement=0;
        titre= "";
    }
    public Document(int num,String titre){
        this.numEnregistrement=num;
        this.titre= titre;
    }

	public int getNumEnregistrement() {
		return numEnregistrement;
	}

	public void setNumEnregistrement(int numEnregistrement) {
		this.numEnregistrement = numEnregistrement;
	}

	public String getTitre() {
		return titre;
	}

	public void setTitre(String titre) {
		this.titre = titre;
	}
	@Override
	public String toString() {
		return "Document [numEnregistrement=" + numEnregistrement + ", titre=" + titre + ", getNumEnregistrement()="
				+ getNumEnregistrement() + ", getTitre()=" + getTitre() + ", getClass()=" + getClass() + ", hashCode()="
				+ hashCode() + "]";
	}
	
}
